sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -191,25 +204,23 @@ internal static (string FormattedError, string? ErrorCode, IDictionary<string, s
             return (formatMessage, error?.Code, data);
         }
 
-        /// <summary>
-        /// This is intentionally sync-only as it will only be called by the ctor.
-        /// </summary>
-        /// <param name="response"></param>
-        /// <returns></returns>
-        internal static string? ReadContent(Response response)
+        private static void BufferResponseIfNeeded(Response response)
         {
-            string? content = null;
-
-            if (response.ContentStream != null &&
-                ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out var encoding))
+            // Buffer into a memory stream if not already buffered
+            if (response.ContentStream is null or MemoryStream)
```

Comment:
This seems rather specific. Should we consider a check on `CanSeek`, or is the concern here that anything but a `MemoryStream` could be disposed?

Reply:
The intent was to verify that the message has already been buffered. Does CanSeek guarantee this? I wasn't sure about that.